### PR TITLE
refactor(unicorn): adapt to new API

### PR DESCRIPTION
**Pull request by antmat**
adapt to https://github.com/3Hren/cocaine-core/pull/76

### CODE REVIEW

File: unicorn/include/cocaine/detail/future.hpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+template<class T>
+std::future<T> make_ready_future(T value) {
+    std::promise<T> promise;
+    promise.set_value(std::move(value));
+    return promise.get_future();
+}
+
+template<class T>
+std::future<T> make_exceptional_future(std::error_code ec) {
```
Comment (3Hren) [inline unicorn/include/cocaine/detail/future.hpp]: Reuse your own function, eh?

File: unicorn/include/cocaine/detail/future.hpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    std::promise<T> promise;
+    auto e = std::make_exception_ptr(std::system_error(std::move(ec)));
+    promise.set_exception(e);
+    return promise.get_future();
+};
+
+template<class T>
+std::future<T> make_exceptional_future(std::error_code ec, std::string msg) {
```
Comment (3Hren) [inline unicorn/include/cocaine/detail/future.hpp]: Here too.

**Comment by 3Hren** [comment]
Ping

**Comment by antmat** [comment]
updated

**Pull request merged**